server/fleet/apple_mdm.go

```diff
@@ -432,6 +432,17 @@ type HostMDMAppleProfile struct {
 	Detail        string                  `db:"detail" json:"detail"`
 }
 
+func (p HostMDMAppleProfile) IgnoreMDMClientError() bool {
+	switch p.OperationType {
+	case MDMAppleOperationTypeRemove:
+		switch {
+		case strings.Contains(p.Detail, "MDMClientError (89)"):
```

Comment:
@roperzh, do you think we need to do more than simple string matching here?

Reply:
yes, but I also think it's fine for now given the circumstances.

nanomdm actually gives us the error code and the domain (which we're actually using to build `p.Detail`) we should use that instead, and probably have an enumeration (similarly to eg `http.StatusOK`)

Reply:
Agreed, I'm thinking that we could handle some of those as part of the refactor for https://github.com/fleetdm/fleet/issues/10122